overlord/snapshotstate/backend/backend_test.go

```diff
@@ -556,16 +556,110 @@ func (s *snapshotSuite) TestList(c *check.C) {
 	}
 }
 
+func (s *snapshotSuite) TestReadSnapshotYamlOpenFails(c *check.C) {
+	var returnedError error
+	defer backend.MockOsOpen(func(string) (*os.File, error) {
+		return nil, returnedError
+	})()
+
+	info := &snap.Info{SideInfo: snap.SideInfo{RealName: "hello-snap", Revision: snap.R(42), SnapID: "hello-id"}}
+
+	// Try a generic error, this is reported as such
+	returnedError = errors.New("Some error")
+	opts := &backend.AddDirToZipOptions{}
+	err := backend.ReadSnapshotYaml(info, opts)
+	c.Check(err, check.ErrorMatches, "Some error")
+
+	// But if the file is not found, that's just a nil error
+	returnedError = os.ErrNotExist
+	err = backend.ReadSnapshotYaml(info, opts)
+	c.Check(err, check.IsNil)
+}
+
+func (s *snapshotSuite) TestReadSnapshotYamlFailures(c *check.C) {
+	var manifestFile *os.File
+	defer backend.MockOsOpen(func(string) (*os.File, error) {
+		manifestFile.Seek(0, os.SEEK_SET)
+		return manifestFile, nil
+	})()
+
+	info := &snap.Info{SideInfo: snap.SideInfo{RealName: "hello-snap", Revision: snap.R(42), SnapID: "hello-id"}}
+
+	for _, testData := range []struct {
+		contents      string
+		expectedError string
+	}{
+		{
+			"", "cannot read snapshot manifest: EOF",
+		},
+		{
+			"invalid", "cannot read snapshot manifest: yaml: unmarshal errors:\n.*",
+		},
+		{
+			"exclude:\n  - /home/ubuntu", "snapshot exclude path must start with one of.*",
+		},
+		{
+			"exclude:\n  - $SNAP_COMMON_STUFF", "snapshot exclude path must start with one of.*",
+		},
+		{
+			"exclude:\n  - $SNAP_DATA/../../meh", "snapshot exclude path not clean.*",
+		},
+	} {
+		var err error
+		manifestFile, err = ioutil.TempFile("", "snapshots.*.yaml")
```

Comment:
Using `c.MkDir()` here is shorter because there is no explicit cleanup needed with that